compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -769,6 +765,42 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
             };
             let captures = format!("captures data from {}", arg);
 
+            if fn_returns.is_empty() {
+                let fn_returns_alias = self
+                    .infcx
+                    .tcx
+                    .is_suitable_region(f)
+                    .map(|r| {
```

Comment:
`flat_map`?

Reply:
Thanks for reviewing. Not sure how to easily use `flat_map` for `is_suitable_region(f)`'s return type `Option<FreeRegionInfo>`. Got following error if simply changing map() to flat_map(): 

error[E0599]: `Option<FreeRegionInfo>` is not an iterator
   --> compiler/rustc_borrowck/src/diagnostics/region_errors.rs:773:22
    |
773 |                     .flat_map(|r| {
    |                      ^^^^^^^^ `Option<FreeRegionInfo>` is not an iterator
    |
   ::: /home/ychen/rust_8.15/rust/library/core/src/option.rs:518:1
    |
518 | pub enum Option<T> {
    | ------------------ doesn't satisfy `Option<FreeRegionInfo>: Iterator`
    |
    = note: the following trait bounds were not satisfied:
            `Option<FreeRegionInfo>: Iterator`
            which is required by `&mut Option<FreeRegionInfo>: Iterator`

Reply:
ah, that requires `and_then`, then. It's annoying that the same operation has a different name on `Option` than on `Iterator` :shrug:

Reply:
Made minor modifications of making use of and_then() and adding comments